### ISSUE TITLE
type-annotation-spacing doesn’t work for function typedefs

### ISSUE THREAD

**Issue by j-f1** [nzakas/eslint-plugin-typescript#41]
This throws an error when it shouldn’t:

```ts
type Foo = (bar: string) => string
//                      ^
// Unexpected space before the colon (typescript/type-annotation-spacing) 
```

Maybe there should be a separate set of options for the `:` annotation and the `=>` annotation, with the default being a space after but not before the `:` and a space before and after the `=>`.

**Comment by weirdpattern** [comment]
I will take a look, thanks for reporting this!

**Comment by weirdpattern** [comment]
I'm adding an overrides option to the configuration that will allow this.
Working on the test, I will submit a new PR tonight
